rosbag2/src/rosbag2/writer.cpp

```diff
@@ -34,19 +37,31 @@ Writer::Writer(
   storage_(nullptr),
   metadata_io_(std::move(metadata_io)),
   converter_(nullptr),
-  max_bagfile_size_(rosbag2_storage::storage_interfaces::MAX_BAGFILE_SIZE_NO_SPLIT)
+  max_bagfile_size_(rosbag2_storage::storage_interfaces::MAX_BAGFILE_SIZE_NO_SPLIT),
+  topics_names_to_info_({}),
+  metadata_()
 {}
 
 Writer::~Writer()
 {
   if (!uri_.empty()) {
-    metadata_io_->write_metadata(uri_, storage_->get_metadata());
+    finalize_metadata();
+    metadata_io_->write_metadata(uri_, metadata_);
   }
 
   storage_.reset();  // Necessary to ensure that the storage is destroyed before the factory
   storage_factory_.reset();
 }
 
+void Writer::init_metadata()
+{
+  metadata_ = rosbag2_storage::BagMetadata{};
```

Comment:
once again, isn't that line redundant to line 42 in the initializer list? Or is `init_metadata` called multiple times to reset the struct?

Reply:
it is redundant, but it also indicates that `metadata_` is zeroed out within `init_metadata()`